src/spec/cli.cr

```diff
@@ -0,0 +1,147 @@
+require "./formatter"
+require "option_parser"
+
+module Spec
+  # :nodoc:
+  class CLI
+    # :nodoc:
+    class Options
+      property formatters = Array(Spec::Formatter).new
+      property locations = Array({String, Int32}).new
+      property default_formatter : Spec::Formatter?
+      property? fail_fast : Bool = false
+      property line : Int32?
+      property? no_color : Bool = false
+      property pattern : String?
+      property slowest : Int32?
+    end
+
+    getter options : Options
+
+    private getter argv : Array(String)
+    private getter parser = OptionParser.new
```

Comment:
Again, why does this need to be stored in an ivar? It is just needed in `prepare`. You could simply initialize it there and call `setup(parser)` (maybe renamed to `setup_parser(parser)`).